pkgs/development/mobile/androidenv/deploy-androidpackages.nix

```diff
@@ -0,0 +1,58 @@
+{stdenv, lib, unzip, mkLicenses}:
+{packages, os ? null, buildInputs ? [], patchesInstructions ? {}, meta ? {}, ...}@args:
+
+let
+  extraParams = removeAttrs args [ "packages" "os" "buildInputs" "patchesInstructions" ];
+  sortedPackages = builtins.sort (x: y: builtins.lessThan x.name y.name) packages;
+in
+stdenv.mkDerivation ({
+  pname = lib.concatMapStringsSep "-" (package: package.name) sortedPackages;
+  version = lib.concatMapStringsSep "-" (package: package.revision) sortedPackages;
+  src = map (package:
+    if os != null && builtins.hasAttr os package.archives then package.archives.${os} else package.archives.all
+  ) packages;
+  buildInputs = [ unzip ] ++ buildInputs;
+  preferLocalBuild = true;
+
+  unpackPhase = ''
+    buildDir=$PWD
+    i=0
+    for srcArchive in $src; do
+      extractedZip="extractedzip-$i"
+      i=$((i+1))
+      cd "$buildDir"
+      mkdir "$extractedZip"
+      cd "$extractedZip"
+      unpackFile "$srcArchive"
+    done
```

Comment:
We should use nix to generate this

Reply:
I love that idea, but have no clue how! can you give me a link to a source that unpacked a list of `src`, so I can learn and apply it here?

Reply:
I think this is the best example how to do that I could find in the last 5 minutes https://github.com/NixOS/nixpkgs/blob/327790ade72ddb53c1824dff9bd47bce33ccb4cc/nixos/modules/services/networking/dnscache.nix#L11-L13

Reply:
Thank you, but I think I didn't explain my problem very well. I cannot get information in the `$src`, which are the paths to downloaded packages, so I could iterate on them. How can I get the list of downloaded packages in Nix?

Reply:
src is for a single src
use srcs and it should work without unpackPhase
https://github.com/NixOS/nixpkgs/blob/39bc9634f64232e817ddc910f5b683c146f2b454/pkgs/stdenv/generic/setup.sh#L923-L926

Reply:
> How can I get the list of downloaded packages in Nix

what exactly is the output you want? i thin structuredAttrs will be useful here but its not in the master branch yet https://github.com/NixOS/nixpkgs/pull/175649 because then the `packages` could be passed to the builder in json

Reply:
Thanks @Artturin, apparently the block you mentioned that iterates on `$srcs` is in `unpackPhase`, which I didn't use! I used `unpackFile` because I needed to have control on the directories that it extracts the packages. Am I missing something here?! Also I remember I couldn't make it work with `$srcs`, however, this solution with `$src` definitely works as expected. Therefore, I'll resolve this conversation.